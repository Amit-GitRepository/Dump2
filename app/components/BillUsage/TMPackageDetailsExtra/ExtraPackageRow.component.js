import moment from '../../../utils/moment.util';
import PropTypes from 'prop-types';
import React, {Component} from 'react';
import result from 'lodash/result';
import styles from './ExtraPackageRow.style';
import {bytesToSize, precisionFormatter} from '../../../utils/common.util';
import {colors} from '../../../themes/constants.styles';
import {ICON_MAP, PROGRESS_BAR, QUOTA_TYPE, QUOTA_VALUE} from '../../../config/billUsage.config';
import {ISText, ProgressBar} from '../../Shared';
import {translate} from '../../../language/i18n/helper';
import {View} from 'react-native';

class ExtraPackageRow extends Component {
  render () {
    const {packageRowItem, index, packageType, length, hideSeparator, isPrepaid} = this.props;
    const {quotaType, packageCode, endDate, packageName, subQuotaType} = packageRowItem;

    let remainingQuota = packageRowItem.remainingQuota;
    let remainingQuotaUnit = packageRowItem.remainingQuotaUnit;
    let initialQuotaUnit = packageRowItem.initialQuotaUnit;
    let initialQuota = packageRowItem.initialQuota;
    if (packageRowItem.fup) { // When FUP present: display FUP values
      remainingQuota = packageRowItem.fup.remainingQuota;
      remainingQuotaUnit = packageRowItem.fup.remainingQuotaUnit;
      initialQuotaUnit = packageRowItem.fup.initialQuotaUnit;
      initialQuota = packageRowItem.fup.initialQuota;
    }

    const packageDisplayName =  packageName ? packageName : packageCode;
    const valueType = (isPrepaid) ? translate('BILLS_USAGE_REMAINING') : translate('BILLS_USAGE_USED');
    const title = (subQuotaType === QUOTA_TYPE.SOCIAL) ? result(ICON_MAP[subQuotaType], 'titleText', '') : result(ICON_MAP[quotaType], 'titleText', '');

    const isUnlimited = initialQuota === QUOTA_VALUE.UNLIMITED || remainingQuota === QUOTA_VALUE.UNLIMITED;
    const initialValueUnit = quotaType === QUOTA_TYPE.VOICE ? translate('BILLS_USAGE_MIN') : initialQuotaUnit;
    const isData = quotaType === QUOTA_TYPE.DATA || quotaType === QUOTA_TYPE.WIFI;

    let value = remainingQuota;
    let unit = quotaType === QUOTA_TYPE.VOICE ? translate('BILLS_USAGE_MIN') : remainingQuotaUnit;
    if (!isPrepaid && !isUnlimited && initialQuota) {
      if (isData) {
        const initialQuotaInBytes = bytesToSize(initialQuota, initialQuotaUnit, 'Bytes');
        const remainingQuotaInBytes = bytesToSize(remainingQuota, remainingQuotaUnit, 'Bytes');
        const usedQuotaInBytes = initialQuotaInBytes.value - remainingQuotaInBytes.value;
        const usedQuota = bytesToSize(usedQuotaInBytes, 'Bytes');
        value = usedQuota.value;
        unit = usedQuota.unit;
      } else {
        value = initialQuota - remainingQuota;
      }
    }

    const percentageValue = (isUnlimited || !initialQuota) ? PROGRESS_BAR.DEFAULT_FILL : (value * 100) / initialQuota;
    const initialValue = precisionFormatter(initialQuota, isData);// showPrecision if data values
    value = precisionFormatter(value, isData);

    return (
      <View style={[styles.container, (length - 1 === index && !hideSeparator) ? {} : styles.separator]}>
        <View style={styles.titleContainer}>
          <ISText type='BOLD' style={styles.itemTitle}>{title ? translate(title) : quotaType}
            <ISText type='BOLD' style={styles.itemSubtitle}>{` (${packageType})`}</ISText>
          </ISText>
          <ISText type='BOLD' style={styles.itemDisplayName}>{packageDisplayName}</ISText>
        </View>
        <View style={styles.contentContainer}>
          <View style={styles.rowContainer}>
            <ISText type='SEMIBOLD' style={styles.itemRowTitle}>{valueType}
              {packageRowItem.fup && <ISText type='BOLD' style={styles.itemSubtitle}>{` (${translate('BILLS_USAGE_MAXSPEED')})`}</ISText>}
            </ISText>
            <ISText type='SEMIBOLD' style={styles.consumedValue}>
              {(!isUnlimited) ? value : translate('BILLS_USAGE_UNLIMITED')}
              {(!isUnlimited && unit) && <ISText type='SEMIBOLD' style={styles.consumedValueUnit}>{' ' + unit}</ISText>}
            </ISText>
          </View>
          {!isPrepaid && <View style={styles.rowContainer}>
            <View style={styles.packageRowLeft}>
              <ProgressBar fillColor={colors.SECONDARY_TMOVE} backgroundColor={colors.PRIMARY_BG_SEPARATOR} percentage={percentageValue} />
            </View>
            <View style={styles.packageRowRight}>
              {(initialQuotaUnit && initialQuota) && <ISText type='SEMIBOLD' style={styles.totalValue}>{translate('BILLS_USAGE_OF') + ' ' + initialValue} {!isUnlimited ? initialValueUnit : ''}</ISText>}
            </View>
          </View>}
          {endDate ? <View style={styles.rowContainer}>
            <ISText type='SEMIBOLD' style={styles.itemRowTitle}>{translate('BILLS_USAGE_VALID_UNTIL')}</ISText>
            <ISText type='SEMIBOLD' style={styles.itemRowValue}>{new moment(endDate).format('DD/MM/YYYY')} </ISText>
          </View> : null}
        </View>
      </View>
    );
  }
}

ExtraPackageRow.defaultProps = {
  packageRowItem: {},
  index: -1,
  packageType: '',
  length: 0,
  hideSeparator: 0,
  isPrepaid: false
};

ExtraPackageRow.propTypes = {
  packageRowItem: PropTypes.object,
  index: PropTypes.number,
  packageType: PropTypes.string,
  length: PropTypes.number,
  hideSeparator: PropTypes.number,
  isPrepaid: PropTypes.bool
};

export default ExtraPackageRow;
