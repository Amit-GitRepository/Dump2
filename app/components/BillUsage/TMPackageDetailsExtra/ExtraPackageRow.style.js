import {colors, fonts} from '../../../themes/constants.styles';

export default {
  container: {
    flex: 1,
    paddingBottom: 10
  },
  separator: {
    borderBottomWidth: 1,
    borderBottomColor: colors.PRIMARY_BG_SEPARATOR,
    marginBottom: 10
  },
  titleContainer: {
    flexDirection: 'column',
    width: '80%',
    paddingBottom: 5
  },
  contentContainer: {
    flexDirection: 'column'
  },
  rowContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'flex-end'
  },
  packageRowLeft: {
    width: '65%',
    paddingBottom: 5
  },
  packageRowRight: {
    width: '35%',
    paddingBottom: 5
  },
  itemTitle: {
    fontSize: fonts.FONT_SIZE_MEDIUM
  },
  itemSubtitle: {
    color: colors.PRIMARY_SUBTEXT,
    fontSize: fonts.FONT_SIZE_SMALL
  },
  itemDisplayName: {
    color: colors.PRIMARY_SUBTEXT
  },
  itemRowTitle: {
    fontSize: fonts.FONT_SIZE_NORMAL
  },
  itemRowValue: {
    fontSize: fonts.FONT_SIZE_NORMAL,
    color: colors.PRIMARY_SUBTEXT
  },
  consumedValue: {
    fontSize: fonts.FONT_SIZE_LARGE,
    marginBottom: -6
  },
  consumedValueUnit: {
    fontSize: fonts.FONT_SIZE_SMALL
  },
  totalValue: {
    color: colors.PRIMARY_SUBTEXT,
    marginBottom: -5,
    alignSelf: 'flex-end'
  },
  validityText: {
    color: colors.PRIMARY_SUBTEXT,
    fontSize: fonts.FONT_SIZE_SMALL
  }
};
