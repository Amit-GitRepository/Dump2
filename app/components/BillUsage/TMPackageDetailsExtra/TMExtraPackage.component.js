import ExtraPackageRow from './ExtraPackageRow.component';
import HeaderWithCaret from '../BillUsage/BillUsageShared/HeaderWithCaret.component';
import PropTypes from 'prop-types';
import React, {Component} from 'react';
import styles from './TMExtraPackage.style';
import {colors} from '../../../themes/constants.styles';
import {ISText, WrapperCard} from '../../Shared';
import {partition} from 'lodash';
import {ScrollView, View} from 'react-native';
import {translate} from '../../../language/i18n/helper';

class TMExtraPackage extends Component {

  render () {
    const {extraPackage, mainPackage, isPrepaid} = this.props;

    const title = extraPackage.length > 0  ? `(${extraPackage.length} ${translate('BILLS_USAGE_EXTRA_PACKAGES')})` : null;
    const filteredExtraPackages = partition(extraPackage, (item) => !item.isActive);
    const filteredMainPackage = partition(mainPackage, (item) => !item.isActive);

    const activeMainPackageContent = filteredMainPackage[0].map((packageRowItem, index) =>
      <ExtraPackageRow key={index} packageRowItem={packageRowItem} index={index} packageType={translate('BILLS_USAGE_MAIN')} length = {filteredMainPackage[0].length} hideSeparator={filteredExtraPackages[0].length} isPrepaid={isPrepaid}/>
    );
    const activeExtraPackageContent = filteredExtraPackages[0].map((packageRowItem, index) =>
      <ExtraPackageRow key={index} packageRowItem={packageRowItem} index={index} packageType={translate('BILLS_USAGE_EXTRA')} length = {filteredExtraPackages[0].length} isPrepaid={isPrepaid}/>
    );
    const inactiveMainPackageContent = filteredMainPackage[1].map((packageRowItem, index) =>
      <ExtraPackageRow key={index} packageRowItem={packageRowItem} index={index} packageType={translate('BILLS_USAGE_MAIN')} length = {filteredMainPackage[1].length} hideSeparator={filteredExtraPackages[1].length} isPrepaid={isPrepaid}/>
    );
    const inactiveExtraPackageContent = filteredExtraPackages[1].map((packageRowItem, index) =>
      <ExtraPackageRow key={index} packageRowItem={packageRowItem} index={index} packageType={translate('BILLS_USAGE_EXTRA')} length = {filteredExtraPackages[1].length} isPrepaid={isPrepaid}/>
    );

    const hasActivePackages = !!(filteredMainPackage[0].length || filteredExtraPackages[0].length);
    const hasInactivePackages = !!(filteredMainPackage[1].length || filteredExtraPackages[1].length);

    return (
      <ScrollView style={styles.scrollContainer} contentContainerStyle={styles.scrollContentContainer}>
        {isPrepaid && title && <View style={styles.titleContainer}>
          <ISText type='BOLD' style={styles.titleText}>{title}</ISText>
        </View>}
        {hasActivePackages && <WrapperCard style={styles.activeContainer}>
          <HeaderWithCaret headerColor={colors.PRIMARY_SUBMENU} caretShadow={true} iconSize={32}>
            <ISText type='BOLD' style={styles.cardHeaderText}>{translate('EXTRA_PACKAGE_ACTIVE')}</ISText>
          </HeaderWithCaret>
          <View style={styles.cardContent}>
            {activeMainPackageContent}
            {activeExtraPackageContent}
          </View>
        </WrapperCard>}
        {hasInactivePackages && <WrapperCard style={styles.inactiveContainer}>
          <HeaderWithCaret headerColor={colors.PRIMARY_DISABLED_BG_TEXT} caretShadow={true} iconSize={32}>
            <ISText type='BOLD' style={styles.cardHeaderText}>{translate('EXTRA_PACKAGE_INACTIVE')}</ISText>
          </HeaderWithCaret>
          <View style={styles.cardContent}>
            {inactiveMainPackageContent}
            {inactiveExtraPackageContent}
          </View>
        </WrapperCard>}
      </ScrollView>
    );
  }
}

TMExtraPackage.defaultProps = {
  extraPackage: [],
  mainPackage: [],
  isPrepaid: false
};

TMExtraPackage.propTypes = {
  extraPackage: PropTypes.array,
  mainPackage: PropTypes.array,
  isPrepaid: PropTypes.bool
};

export default TMExtraPackage;
