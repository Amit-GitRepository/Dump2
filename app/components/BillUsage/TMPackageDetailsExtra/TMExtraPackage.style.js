import {colors, fonts} from '../../../themes/constants.styles';

export default {
  scrollContainer: {
    marginTop: -10
  },
  scrollContentContainer: {
    paddingTop: 10
  },
  cardHeaderText: {
    color: colors.PRIMARY_BG_TEXT_CONTRAST,
    textAlign: 'center',
    fontSize: fonts.FONT_SIZE_MEDIUM
  },
  cardContent: {
    paddingHorizontal: 15
  },
  activeContainer: {
    marginTop: 10,
    marginHorizontal: 16
  },
  inactiveContainer: {
    marginVertical: 16,
    marginHorizontal: 16
  },
  separator: {
    borderBottomWidth: 1,
    borderBottomColor: colors.PRIMARY_BG_SEPARATOR,
    paddingBottom: 5,
    marginBottom: 5
  },
  titleContainer: {
    justifyContent: 'center',
    alignItems: 'center',
    paddingVertical: 20
  },
  titleText: {
    fontSize: fonts.FONT_SIZE_MEDIUM
  }
};
