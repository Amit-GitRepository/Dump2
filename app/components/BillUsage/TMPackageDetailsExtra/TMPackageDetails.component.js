import DetailCardRow from '../BillUsage/BillUsageShared/DetailCardRow.component';
import HeaderWithCaret from '../BillUsage/BillUsageShared/HeaderWithCaret.component';
import PackageDetailHeader from '../BillUsage/BillUsageShared/PackageDetailHeader/PackageDetailHeader.component';
import PackageOverview from '../BillUsage/BillUsageShared/PackageOverview/PackageOverview.component';
import PropTypes from 'prop-types';
import React, {Component} from 'react';
import result from 'lodash/result';
import styles from './TMPackageDetails.style';
import {ICON_MAP, QUOTA_TYPE} from '../../../config/billUsage.config';
import {ScrollView, View} from 'react-native';
import {translate} from '../../../language/i18n/helper';
import {WrapperCard} from '../../Shared';

class TMPackageDetails extends Component {
  render () {
    const {pricePlanInfo, mainPackage, billCycle, creditLimit, offerDesc, isPrepaid, isConvergence, phoneNumber} = this.props;
    const allPackages = mainPackage.map((packageDetail, index) => {
      const subType = (packageDetail.subQuotaType === QUOTA_TYPE.SOCIAL) ? packageDetail.subQuotaType : packageDetail.quotaType;
      const type = packageDetail.quotaType;

      let subText = '';
      let initialQuota = packageDetail.initialQuota;
      let initialQuotaUnit = packageDetail.initialQuotaUnit;
      if (packageDetail.fup) { // When FUP present: display FUP values
        subText = translate('BILLS_USAGE_MAXSPEED');
        initialQuota = packageDetail.fup.initialQuota;
        initialQuotaUnit = packageDetail.fup.initialQuotaUnit;
      }
      return (
        type ? <DetailCardRow value={initialQuota} unit={initialQuotaUnit} key={index}
          icon= {result(ICON_MAP[subType], 'titleIconPath', '')} title= {result(ICON_MAP[subType], 'titleText', '')} type={type} showSeparator={!((mainPackage.length - 1 === index) && !creditLimit)} subText={subText}/>
          : null
      );
    });
    const tmPackageDetail =   <View style={styles.container}>
      <PackageOverview title = {pricePlanInfo.description} subTitle = {offerDesc} showSeparator={!!(mainPackage.length || creditLimit)}/>
      {!isPrepaid && allPackages}
      {(!isPrepaid && creditLimit) ? <DetailCardRow value={creditLimit} unit={translate('BILLS_USAGE_THB')} title= 'BILLS_USAGE_CREDIT_LIMIT' showSeparator={false}/> : null}
    </View>;

    return (
      <ScrollView style={styles.scrollContainer} contentContainerStyle={styles.scrollContentContainer}>
        <WrapperCard style={styles.wrapperContainer}>
          <HeaderWithCaret>
            <PackageDetailHeader isConvergence={isConvergence} name={translate('BILLS_USAGE_TRUEMOVE')} phoneNumber={phoneNumber}
              endDate={result(billCycle, 'closeDay', 0)} isPrepaid={isPrepaid}/>
          </HeaderWithCaret>
          {tmPackageDetail}
        </WrapperCard>
      </ScrollView>
    );
  }
}

TMPackageDetails.defaultProps = {
  pricePlanInfo: {},
  mainPackage: [],
  billCycle: {},
  creditLimit: '',
  offerDesc: '',
  isPrepaid: false,
  isConvergence: false,
  phoneNumber: ''
};

TMPackageDetails.propTypes = {
  pricePlanInfo: PropTypes.object,
  mainPackage: PropTypes.array,
  billCycle: PropTypes.object,
  creditLimit: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  offerDesc: PropTypes.string,
  isPrepaid: PropTypes.bool,
  isConvergence: PropTypes.bool,
  phoneNumber: PropTypes.string
};

export default TMPackageDetails;
