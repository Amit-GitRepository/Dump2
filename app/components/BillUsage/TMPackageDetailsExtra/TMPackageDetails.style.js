import {colors, fonts} from '../../../themes/constants.styles';

export default {
  scrollContainer: {
    marginTop: -10
  },
  scrollContentContainer: {
    paddingTop: 10,
    alignItems: 'center'
  },
  container: {
    paddingBottom: 10,
    paddingHorizontal: 10
  },
  creditItemText: {
    textAlign: 'center',
    color: colors.PRIMARY_SUBTEXT
  },
  creditItemCost: {
    fontSize: fonts.FONT_SIZE_XXL,
    color: colors.PRIMARY_SUBTEXT
  },
  wrapperContainer: {
    marginVertical: 10,
    marginHorizontal: 8,
    alignSelf: 'stretch'
  }
};
