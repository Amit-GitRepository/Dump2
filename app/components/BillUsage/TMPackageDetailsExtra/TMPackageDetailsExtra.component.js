
import PropTypes from 'prop-types';
import React, {Component} from 'react';
import styles from './TMPackageDetailsExtra.style';
import TMExtraPackage from './TMExtraPackage.component';
import TMPackageDetails from './TMPackageDetails.component';
import {CaretTabs} from '../../Shared';
import {translate} from '../../../language/i18n/helper';
import {View} from 'react-native';

class TMPackageDetailsExtra extends Component {
  state = {
    selectedTabIndex: 0
  }

  onTabPress = (selectedTabIndex) => {
    this.setState({selectedTabIndex});
  }
  render () {
    const {pricePlanInfo, isPrepaid, mainPackage,  extraPackage, billCycle, creditLimit, offerDesc, phoneNumber} = this.props;
    const {selectedTabIndex} = this.state;
    const headerList = [{
      text: translate('PACKAGE_DETAIL__TITLE')
    }, {
      text: translate('EXTRA_PACKAGE_TITLE')
    }];

    return (
      <View style={styles.container}>
        {isPrepaid ?  <TMExtraPackage mainPackage={mainPackage} extraPackage={extraPackage} isPrepaid={isPrepaid}/> : <CaretTabs onTabPress={this.onTabPress} selectedIndex={selectedTabIndex} headerList={headerList} headerContainerStyle={styles.caretTabContainer} bodyStyle={styles.bodyStyle}>
          {selectedTabIndex === 0 && <TMPackageDetails mainPackage={mainPackage} pricePlanInfo={pricePlanInfo} billCycle={billCycle} creditLimit={creditLimit} offerDesc={offerDesc} isPrepaid={isPrepaid} phoneNumber={phoneNumber}/>}
          {selectedTabIndex === 1 && <TMExtraPackage mainPackage={mainPackage} extraPackage={extraPackage} isPrepaid={isPrepaid}/>}
        </CaretTabs>}
      </View>
    );
  }
}

TMPackageDetailsExtra.defaultProps = {
  pricePlanInfo: {},
  mainPackage: [],
  extraPackage: [],
  billCycle: {},
  creditLimit: '',
  offerDesc: '',
  isPrepaid: false,
  phoneNumber: ''
};

TMPackageDetailsExtra.propTypes = {
  pricePlanInfo: PropTypes.object,
  mainPackage: PropTypes.array,
  extraPackage: PropTypes.array,
  billCycle: PropTypes.object,
  creditLimit: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  offerDesc: PropTypes.string,
  isPrepaid: PropTypes.bool,
  phoneNumber: PropTypes.string
};

export default TMPackageDetailsExtra;
