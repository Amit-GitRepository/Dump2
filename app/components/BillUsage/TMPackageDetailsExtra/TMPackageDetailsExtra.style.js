import {colors} from '../../../themes/constants.styles';

export default {
  container: {
    paddingVertical: 10,
    flex: 1
  },
  caretTabContainer: {
    paddingHorizontal: 20
  },
  bodyStyle: {
    flex: 1,
    backgroundColor: colors.TRANSPARENT
  },
  headerStyle: {
    padding: 10
  },
  headerColor: colors.SECONDARY_TMOVE
};
