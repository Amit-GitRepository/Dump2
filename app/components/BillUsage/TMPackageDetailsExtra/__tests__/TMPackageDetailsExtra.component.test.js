import React from 'react';
import ReactSixteenAdapter from 'enzyme/build/adapters/ReactSixteenAdapter';
import renderer from 'react-test-renderer';
import TMPackageDetailsExtra from '../TMPackageDetailsExtra.component'; // Enzyme adapter for React 16
import {configure} from 'enzyme';

configure({adapter: new ReactSixteenAdapter()});

describe('TMPackageDetailsExtra component', () => {
  it('renders correctly', () => {
    const component = renderer.create(<TMPackageDetailsExtra />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
