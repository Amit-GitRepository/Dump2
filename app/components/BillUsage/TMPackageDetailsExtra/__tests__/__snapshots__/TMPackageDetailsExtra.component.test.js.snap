// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TMPackageDetailsExtra component renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingVertical": 10,
    }
  }
>
  <View
    accessibilityLabel={undefined}
    style={
      Object {
        "alignSelf": "stretch",
        "flexGrow": 1,
      }
    }
    testID={undefined}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-start",
            "flexDirection": "row",
            "zIndex": 2,
          },
          Object {
            "paddingHorizontal": 20,
          },
        ]
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderTopLeftRadius": 10,
            "borderTopRightRadius": 10,
            "flex": 1,
            "marginRight": 5,
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "backgroundColor": "#F93E60",
                  "borderTopLeftRadius": 10,
                  "borderTopRightRadius": 10,
                  "height": 42,
                  "overflow": "hidden",
                  "paddingHorizontal": 10,
                  "paddingTop": 7,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Object {
                    "alignSelf": "center",
                    "backgroundColor": "transparent",
                    "color": "#FFFFFF",
                    "fontFamily": "SukhumvitSet-Bold",
                    "fontSize": 18,
                    "textAlign": "center",
                  }
                }
              >
                Package Detail
              </Text>
            </View>
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#F93E60",
                    "fontSize": 30,
                  },
                  Array [
                    Object {
                      "alignSelf": "center",
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "bottom": 12,
                      "marginBottom": -12,
                    },
                    Object {},
                  ],
                  Object {
                    "fontFamily": "iservice",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              Ó§Å
            </Text>
          </View>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderTopLeftRadius": 10,
            "borderTopRightRadius": 10,
            "flex": 1,
            "marginRight": 0,
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "transparent",
                "borderRadius": 0,
                "height": 42,
                "overflow": "hidden",
                "paddingBottom": 0,
                "paddingTop": 0,
              }
            }
          >
            <View
              style={Object {}}
            >
              <Image
                getSize={[Function]}
                isBackground={false}
                onLayout={[Function]}
                resizeMode="contain"
                source={
                  Object {
                    "testUri": "../../../app/assets/deselectedTabBg.png",
                  }
                }
                style={
                  Array [
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    Object {
                      "height": undefined,
                      "width": undefined,
                    },
                    undefined,
                  ]
                }
              />
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Object {
                    "alignSelf": "center",
                    "backgroundColor": "transparent",
                    "color": "#000000",
                    "fontFamily": "SukhumvitSet-Bold",
                    "fontSize": 18,
                    "paddingTop": 2,
                    "textAlign": "center",
                  }
                }
              >
                Extra Package
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "borderColor": "#F93E60",
            "borderWidth": 1,
            "marginTop": -19,
            "zIndex": 3,
          },
          Object {
            "borderColor": "#F93E60",
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "#FFFFFF",
            "borderBottomLeftRadius": 7,
            "borderBottomRightRadius": 7,
            "paddingTop": 10,
          },
          Object {
            "backgroundColor": "transparent",
            "flex": 1,
          },
          Object {
            "shadowColor": "#303C46",
            "shadowOffset": Object {
              "height": 2,
              "width": 1,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 1,
          },
        ]
      }
    >
      <RCTScrollView
        contentContainerStyle={
          Object {
            "alignItems": "center",
            "paddingTop": 10,
          }
        }
        style={
          Object {
            "marginTop": -10,
          }
        }
      >
        <View>
          <View
            style={
              Array [
                Object {
                  "marginHorizontal": 8,
                },
                Object {
                  "alignSelf": "stretch",
                  "marginHorizontal": 8,
                  "marginVertical": 10,
                },
              ]
            }
          >
            <View
              elevation={2}
              style={
                Array [
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "borderColor": "#B9C1C9",
                    "borderRadius": 10,
                    "borderWidth": 1,
                    "shadowColor": "#303C46",
                    "shadowOffset": Object {
                      "height": 2,
                      "width": 1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 1,
                  },
                  Object {},
                ]
              }
            >
              <View
                style={undefined}
              >
                <View>
                  <View
                    style={Object {}}
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "borderTopLeftRadius": 10,
                            "borderTopRightRadius": 10,
                            "padding": 10,
                          },
                          Object {},
                          Object {
                            "backgroundColor": "#FBBC1E",
                          },
                        ]
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "flexDirection": "row",
                            "justifyContent": "space-between",
                          }
                        }
                      >
                        <View
                          style={
                            Array [
                              false,
                              Object {},
                            ]
                          }
                        >
                          <Text
                            accessible={true}
                            allowFontScaling={true}
                            ellipsizeMode="tail"
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "transparent",
                                  "color": "#303C46",
                                },
                                Array [
                                  Object {
                                    "color": "#FFFFFF",
                                    "fontSize": 14,
                                    "marginBottom": -5,
                                  },
                                  Object {},
                                ],
                                Object {
                                  "fontFamily": "SukhumvitSet-Bold",
                                },
                              ]
                            }
                          >
                            TrueMove H
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                  <Text
                    accessible={true}
                    allowFontScaling={false}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "#FBBC1E",
                          "fontSize": 30,
                        },
                        Array [
                          Object {
                            "alignSelf": "center",
                            "backgroundColor": "transparent",
                          },
                          Object {
                            "bottom": 12,
                            "marginBottom": -12,
                          },
                          Object {},
                        ],
                        Object {
                          "fontFamily": "iservice",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                      ]
                    }
                  >
                    Ó§Å
                  </Text>
                </View>
              </View>
              <View
                style={
                  Object {
                    "paddingBottom": 10,
                    "paddingHorizontal": 10,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "paddingHorizontal": 5,
                      "paddingTop": 2,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#000000",
                        "fontFamily": "SukhumvitSet-Bold",
                        "fontSize": 16,
                        "textAlign": "center",
                      }
                    }
                  >
                    
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
